x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/WordBoundaryChunker.java

```diff
@@ -55,6 +56,19 @@ public List<String> chunk(String input, ChunkingSettings chunkingSettings) {
         }
     }
 
+    /**
+     * Utility method for testing.
+     * Use the chunk functions that return offsets where possible
+     */
+    List<String> textChunks(String input, int chunkSize, int overlap) {
```

Comment:
Can the tests just validate that the chunk offsets returned are correct? Since the word and sentence chunkers don't return the text is there a reason we need to validate the text returned? 

If there is a reason we need to validate the text, can this be moved to the testing files? I think the logic should be the same across chunkers so it could be added to a shared chunker utility test class.

Reply:
Good point, these methods are only used in tests I've moved them to the test classes. It makes it easier to debug the tests if you are looking at strings rather than offsets so they are useful to have